### DISCUSSION TITLE
With NGINX configured, the airflow web redirect to home page instead of the specific Dag Page after clear.

### DISCUSSION THREAD

**Discussion by yogyang** [apache/airflow#14450] [Q&A] [answered]
After clearing the airflow task/dag (Or marking Failed/Success), the airflow web redirect to the home page instead of the specific Dag Page.
Starting from Airflow 1.10.14.

web server log:
![image](https://user-images.githubusercontent.com/33287576/109121836-bb7e4300-7782-11eb-85da-8e5abc390316.png)

**Comment by turbaszek** [comment]
@yogyang if you think this is a bug please consider creating an bug report:
https://github.com/apache/airflow/issues/new/choose

**Reply by yogyang** [reply]
It's not a bug, just a production Nginx config problem, will update the answer right now.
just write it in case someone will meet this later. It cost me two days to figure it out.

**Comment by yogyang** [comment, marked as answer]
Just create the question in case someone will meet it again.
Cause:
From Airflow 1.10.14, Airflow introduced new codes for redirection location.
https://github.com/apache/airflow/blob/5bfa0f123b39babe1ef66c139e59e452240a6bd7/airflow/www/views.py#L122-L136

And when set up with NGINX, by default, NGINX rewrites the host to $proxy-host, making the `request.host` does not match with input URL parsed netloc`

Fix:
Add `proxy_set_header Host $host;`  in NGINX config.

Refer:
https://docs.nginx.com/nginx/admin-guide/web-server/reverse-proxy/#passing-request-headers

**Reply by springuper** [reply]
I'd strongly recommend we add `logging.warning(...)` before https://github.com/apache/airflow/blob/5bfa0f123b39babe1ef66c139e59e452240a6bd7/airflow/www/views.py#L138 to explicitly notice developers to know the redirection reason.
